rest/diagnostic_doc_api.go

```
@@ -183,20 +187,40 @@ func (h *handler) handleSyncFnDryRun() error {
 func (h *handler) handleImportFilterDryRun() error {
 	docid := h.getQuery("doc_id")
 
-	body, err := h.readDocument()
-	if err != nil {
-		if docid == "" {
-			return fmt.Errorf("Error reading body: %s, no doc id provided for dry run", err)
-		}
+	var importFilterPayload ImportFilterDryRunPayload
+	err := h.readJSONInto(&importFilterPayload)
+	// Only require a valid JSON payload if docid is not provided.
+	// If docid is provided, the sync function will use the document from the bucket, and the payload is optional.
+	if err != nil && docid == "" {
```

Same issue here as on the sync function PR - I don't think we should swallow this error when `docid` is empty since we'd be allowing this endpoint to run on invalid JSON.